### PR TITLE
Hook registration with backend

### PR DESCRIPTION
**Pull request by nhanaa**
Updates:
- Add hooks to connect frontend with backend for registration.

### CODE REVIEW

**Review by GoZaddy — COMMENTED** [review]
Link to register as crimson defense member currently goes to /register/student instead of register/cd-member

**Review by GoZaddy — CHANGES_REQUESTED** [review]
- link to register as cd-member goes to /register/student
- backend returns a 400 status code when registering as cd-member (looks like the request body is missing username?) I'm also not sure why we need the competition id to create crimson defense member accounts but I'll talk to backend about that
- i'm also getting a cors issue when registering as teacher(will also talk to backend)

File: src/hooks/accounts.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+export const useCreateTeacher = (
+  onSuccessFn?: ((data: any, variables: CreateTeacherRequest, context: unknown) => Promise<unknown> | unknown) | undefined,
+  onErrorFn?: ((error: Error, variables: CreateTeacherRequest, context: unknown) => Promise<unknown> | unknown) | undefined
+) : {
+  mutate: UseMutateFunction<any, Error, CreateTeacherRequest, unknown>,
+  isPending: boolean;
+} => {
+  const createTeacher = async (request_body: CreateTeacherRequest) => {
+    const response = await fetch(
+      `${BASE_API_URI}/accounts/teacher/create`,
```
Comment (GoZaddy) [inline src/hooks/accounts.hooks.ts:50]: This should be `/accounts/teachers/create` (plural teachers instead of teacher)

**Comment by nhanaa** [comment]
> * link to register as cd-member goes to /register/student
> * backend returns a 400 status code when registering as cd-member (looks like the request body is missing username?) I'm also not sure why we need the competition id to create crimson defense member accounts but I'll talk to backend about that
> * i'm also getting a cors issue when registering as teacher(will also talk to backend)

Yeah the backend also requires some changes which I made in a separate PR in the backend to work. They should be getting rid of competition id and username.

**Review by GoZaddy — APPROVED** [review]
lgtm!

**Pull request merged**